rootfs/var/lib/dietpi/services/dietpi-wifi-monitor.sh

```diff
@@ -12,12 +12,20 @@
 	[[ $TICKRATE =~ ^[1-9][0-9]*$ ]] || readonly TICKRATE=10
 	GATEWAY=
 
+	# Sleep without subprocess
+	# https://blog.dhampir.no/content/sleeping-without-a-subprocess-in-bash-and-how-to-sleep-forever
+	snore() {
+		local IFS
+		[[ -n "${_snore_fd:-}" ]] || { exec {_snore_fd}<> <(:); } 2>/dev/null
+		read -t "${1:-0}" -u $_snore_fd || :
+	}
```

Comment:
Let's simplify it for our use case:
- If we leave `IFS` at a non-default value elsewhere, we wan't to know this as early as possible through false behaviour instead of gracefully resetting it here. Generally we only temporarily change it and reset it to default afterwards.
- bash conditionals can be much simpler: No need for quotation, no need to replace an empty or undefined variable with an empty string `${var:-}`, `-n` is redundant (and we actively make use of this throughout our code).
- In case there is an error when opening or assigning the file descriptor, we want to see it.
- `read -t ''` and  `read -t 0` behave the same, so no need to replace the empty argument here.
```suggestion
	snore()
	{
		[[ $_snore_fd ]] || exec {_snore_fd}<> <(:)
		read -t "$1" -u "$_snore_fd" || :
	}
```

Reply:
This looks beautiful! To be honest I'm not familiar with bash script that much, so thanks for pointing those points out.
I'll try to test this portion before making a new commit.